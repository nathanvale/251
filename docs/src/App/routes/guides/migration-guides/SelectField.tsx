import React from "react";
import ReactMarkdown from "react-markdown";
import {
  Emphasis,
  List,
  ListItem,
  Stack,
  Strong,
  Text,
  TextLink,
} from "@origin-digital/ods-core";
import { DiffCodeBlock } from "../../../DiffCodeBlock/DiffCodeBlock";
import { Page, PageSection } from "../../../Page/Page";
import {
  baseMigrationGuidesURL,
  ComponentMigrationGuide,
  STYLE_GUIDE_URL,
} from "./consts";

const COMP_NAME = "SelectField";

export const selectFieldMigrationGuide: ComponentMigrationGuide = {
  compName: COMP_NAME,
  route: `${baseMigrationGuidesURL}/${COMP_NAME}`,
  ListItemComp: () => (
    <TextLink href={`${baseMigrationGuidesURL}/${COMP_NAME}`}>
      {COMP_NAME}
    </TextLink>
  ),
  PageComp: () => {
    const sections: PageSection[] = [
      {
        title: "Overview",
        children: (
          <Stack>
            <Text>
              As <Strong>SelectField</Strong> has most of its API in common with
              the <Strong>TextField</Strong> please refer to{" "}
              <TextLink href={`${baseMigrationGuidesURL}/TextField`}>
                TextField migration guide
              </TextLink>{" "}
              first, for shared props. Here we only provide migration guides
              that are unique to <Strong>SelectField</Strong>.
            </Text>
          </Stack>
        ),
      },
      {
        title: "API Changes",
        children: (
          <List>
            <ListItem>
              The <Emphasis>promp</Emphasis> prop is renamed to{" "}
              <Emphasis>displayEmpty</Emphasis>. For improved accessibility the
              new <Strong>SelectField</Strong> also has the{" "}
              <Emphasis>emptyAriaLabel</Emphasis> prop which can provide a
              readable label for the empty option. It is set to{" "}
              <Strong>"None"</Strong> by default.
            </ListItem>
          </List>
        ),
      },
      {
        title: "Diff",
        children: (
          <ReactMarkdown
            renderers={{ code: DiffCodeBlock }}
            source={`\`\`\`diff
<SelectField
  ...
-  prompt={true}
+  displayEmpty={true}
/>

\`\`\``}
          />
        ),
      },
      {
        title: "Previous Implementation",
        children: (
          <List>
            <ListItem>
              <TextLink href={`${STYLE_GUIDE_URL}/${COMP_NAME}`}>
                Origin Style Guide
              </TextLink>
            </ListItem>
          </List>
        ),
      },
    ];

    return <Page title={`${COMP_NAME} Migration Guide`} sections={sections} />;
  },
};

export default selectFieldMigrationGuide;
