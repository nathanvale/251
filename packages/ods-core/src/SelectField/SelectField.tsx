import * as React from "react";
import { SvgIcon, SvgIconProps } from "@material-ui/core";
import { BaseFieldProps, TextFieldBase } from "../TextFieldBase/TextFieldBase";

export type OptionType = {
  value: string;
  label: string;
};

export interface SelectFieldProps extends Omit<BaseFieldProps, "placeholder"> {
  children?:
    | React.ReactElement<"option" | "optgroup">
    | React.ReactElement<"option" | "optgroup">[];
  displayEmpty?: boolean;
  domProps?: React.HTMLAttributes<HTMLInputElement>;
  emptyAriaLabel?: string;
  options?: OptionType[];
}

const IconExpandMore = (props: SvgIconProps) => (
  <SvgIcon {...props}>
    <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
  </SvgIcon>
);

export const SelectField = ({
  children,
  disabled,
  displayEmpty,
  emptyAriaLabel = "None",
  error,
  options,
  ...rest
}: SelectFieldProps) => {
  return (
    <TextFieldBase
      {...rest}
      select={true}
      SelectProps={{
        IconComponent: IconExpandMore,
      }}
      disabled={disabled}
      error={error}
    >
      {displayEmpty && <option aria-label={emptyAriaLabel} value="" />}
      {options
        ? options.map(({ value, label }) => (
            <option value={value} key={value}>
              {label}
            </option>
          ))
        : children}
    </TextFieldBase>
  );
};

SelectField.displayName = "SelectField";
SelectField.defaultProps = {
  variant: "filled",
  size: "medium",
  emptyAriaLabel: "None",
};
