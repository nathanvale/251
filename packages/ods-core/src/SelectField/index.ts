export { SelectField } from "./SelectField";
export type SelectFieldProps = import("./SelectField").SelectFieldProps;
