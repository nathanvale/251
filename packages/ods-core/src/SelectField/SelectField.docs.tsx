import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { removeProps } from "@origin-digital/ods-helpers";
import { Stack } from "../Stack/Stack";
import { Text } from "../Text/Text";
import { Columns } from "../Columns/Columns";
import { Column } from "../Column/Column";
import { Box } from "../Box/Box";
import { baseFieldPropDesc } from "../TextFieldBase/TextFieldBase.docs";
import { SelectField, SelectFieldProps } from "./SelectField";

export const docs: ComponentDocs<SelectFieldProps> = {
  category: "Interaction",
  componentName: "SelectField",
  description: [
    "This component is used when we wanted the user to select an option from a finite set of options",
  ].join(" "),
  propDescriptions: {
    ...removeProps(baseFieldPropDesc("SelectField"), ["placeholder"]),
    children: [
      `An array of <option /> and/or <optgroup /> elements that will be rendered in the dropdown list.`,
      `Use this if you want control over rendering of the option elements. One example of using this`,
      `prop is when we want to group the options into sub-groups. In this case one can fully customise the`,
      `structure of the options children inside the ""select" element via passing down children to SelectField.`,
      `However, the preferred approach for most cases though is to use "options" prop.`,
    ].join(" "),
    options: [
      "An array of objects of the format {value, label} which will represent the option element in the",
      "dropdown of the SelectField. The values must be unique.",
    ].join(" "),
    displayEmpty: [
      `Adds an empty option with "" value at the top of the options.`,
      `aria-label prop for that option is by default set to "None" but could be overridden`,
      `via "emptyAriaLabel" prop.`,
    ].join(" "),
    emptyAriaLabel: [
      `Use this prop to override the default "aria-label" provided for the empty option when "displayEmpty" is set.`,
    ].join(" "),
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <SelectField
          id="select-field1"
          label="Cities"
          options={[
            { value: "syd", label: "Sydney" },
            { value: "mlb", label: "Melbourne" },
            { value: "prt", label: "Perth" },
          ]}
        />
      ),
    },
    additional: [
      {
        label: "Empty option",
        description: [
          `Adds an empty option with "" value at the top of the options.`,
          `aria-label prop for that option is by default set to "None" but could be overridden`,
          `via "emptyAriaLabel" prop.`,
        ].join(" "),
        Code: () => (
          <SelectField
            id="select-field3"
            label="City"
            helperText="City must be selected"
            displayEmpty
            emptyAriaLabel="Empty"
            error
            options={[
              { value: "syd", label: "Sydney" },
              { value: "mlb", label: "Melbourne" },
              { value: "prt", label: "Perth" },
            ]}
          />
        ),
      },
      {
        label: "Handle change",
        description:
          "To handle change listen on onChange and provide the value prop",
        playroom: false,
        Code: () => {
          const [value, setValue] = React.useState<string>();
          return (
            <Stack space="medium">
              <SelectField
                id="select-field4"
                label="City"
                helperText="Please select a city"
                value={value}
                onChange={(e) => setValue(e.target.value)}
                options={[
                  { value: "cbr", label: "Canberra" },
                  { value: "brs", label: "Brisbane" },
                  { value: "adl", label: "Adelaide" },
                ]}
                displayEmpty
              />
              {value && <Text tone="positive">Selected city: {value}</Text>}
            </Stack>
          );
        },
        codeString: `const [value, setValue] = React.useState<string>();
return (<Stack space="medium">
  <SelectField
    id="select-field4"
    label="City"
    helperText="Please select a city"
    value={value}
    onChange={(e) => setValue(e.target.value)}
    options={[
        { value: "cbr", label: "Canberra" },
        { value: "brs", label: "Brisbane" },
        { value: "adl", label: "Adelaide" },
    ]}
    displayEmpty
  />
  {value && <Text tone="positive">Selected city: {value}</Text>}
</Stack>)`,
      },
      {
        label: "How to set width",
        description: [
          "SelectField is always full width. So in order to set its width to a specific value",
          "it should be wrapped in a parent container which has its width set to the desired value.",
          "For most cases you should use Columns/Column as the parent to set the width to match",
          "our design system proportions. In case a pixel perfect width is needed wrap it with a",
          "Box component and set the width on the Box.",
        ].join(" "),
        Code: () => (
          <Stack space="medium">
            <Columns space="medium">
              <Column width="1/3">
                <SelectField
                  id="select-field5"
                  label="State"
                  helperText="Column parent 1/3"
                  options={[
                    { value: "nsw", label: "New South Wales" },
                    { value: "vic", label: "Victoria" },
                  ]}
                />
              </Column>
              <Column width="2/3">
                <SelectField
                  id="select-field6"
                  label="City"
                  helperText="Column parent 2/3"
                  options={[
                    { value: "syd", label: "Sydney" },
                    { value: "mlb", label: "Melbourne" },
                  ]}
                />
              </Column>
            </Columns>
            <Box style={{ width: "8ch" }}>
              <SelectField
                id="select-field7"
                label="Size"
                helperText="Box parent"
                options={[
                  { value: "sm", label: "Small" },
                  { value: "lg", label: "Large" },
                ]}
              />
            </Box>
          </Stack>
        ),
      },
      {
        label: "size",
        description:
          "SelectField could be compact taking less vertical space (48px instead of 56px)",
        Code: () => (
          <Columns space="medium">
            <Column>
              <SelectField
                id="select-field8"
                label="Compact"
                size="small"
                options={[
                  { value: "syd", label: "Sydney" },
                  { value: "mlb", label: "Melbourne" },
                ]}
              />
            </Column>
            <Column>
              <SelectField
                id="select-field9"
                label="Normal"
                options={[
                  { value: "syd", label: "Sydney" },
                  { value: "mlb", label: "Melbourne" },
                ]}
              />
            </Column>
          </Columns>
        ),
      },
      {
        label: "Customised options",
        description: [
          "There are cases where just providing an array of options is not enough. For example",
          "if we want to group the options into sub-groups. In this case one can fully customise the",
          "structure of the options children inside the `select` element via passing down children to SelectField.",
        ].join(" "),
        Code: () => (
          <SelectField
            id="select-field10"
            label="Cities"
            displayEmpty
            emptyAriaLabel="Nothing"
          >
            <optgroup label="NSW">
              <option value="sydney">Sydney</option>
              <option value="newcastle">New Castle</option>
            </optgroup>
            <optgroup label="VIC">
              <option value="melbourne">Melbourne</option>
              <option value="bendigo">Bendigo</option>
            </optgroup>
          </SelectField>
        ),
      },
      {
        label: "Hiding helperText space",
        description: [
          "You can set a flag to hide the space that might be used by helperText.",
          "This allows us have a better user experience by not dynamically changing",
          "the height of the SelectField which means the other components under it won't shift up and down.",
        ].join(" "),
        Code: () => (
          <Stack>
            <SelectField
              id="select-field11"
              label="Keeps space"
              hideHelperTextSpace
              options={[{ value: "syd", label: "Sydney" }]}
            />
            <SelectField
              id="select-field12"
              label="Doesn't keep space"
              options={[{ value: "syd", label: "Sydney" }]}
            />
            <SelectField
              id="select-field13"
              label="Type here"
              helperText="This will be the error"
              options={[{ value: "syd", label: "Sydney" }]}
            />
          </Stack>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Basic",
      Code: () => (
        <SelectField
          id="select-field"
          label="label"
          helperText="details"
          options={[
            { value: "syd", label: "Sydney" },
            { value: "mlb", label: "Melbourne" },
          ]}
        />
      ),
    },
    {
      label: "With Empty option",
      Code: () => (
        <SelectField
          id="select-field"
          label="label"
          helperText="details"
          displayEmpty
          emptyAriaLabel="None"
          options={[
            { value: "syd", label: "Sydney" },
            { value: "mlb", label: "Melbourne" },
          ]}
        />
      ),
    },
    {
      label: "Error",
      Code: () => (
        <SelectField
          id="select-field"
          label="label"
          helperText="details"
          error
          options={[
            { value: "syd", label: "Sydney" },
            { value: "mlb", label: "Melbourne" },
          ]}
        />
      ),
    },
    {
      label: "Custom options",
      Code: () => (
        <SelectField
          id="select-field10"
          label="Cities"
          displayEmpty
          emptyAriaLabel="Nothing"
        >
          <optgroup label="NWS">
            <option value="sydney">Sydney</option>
            <option value="newcastle">New Castle</option>
          </optgroup>
          <optgroup label="VIC">
            <option value="melbourne">Melbourne</option>
            <option value="bendigo">Bendigo</option>
          </optgroup>
        </SelectField>
      ),
    },
    {
      label: "Receive changes",
      Code: () => (
        <SelectField
          id="select-field"
          label="label"
          helperText="details"
          onChange={(e) => {
            /* eslint-disable-next-line no-console */
            console.log("Change received: ", e.target.value);
          }}
          displayEmpty
          options={[
            { value: "syd", label: "Sydney" },
            { value: "mlb", label: "Melbourne" },
          ]}
        />
      ),
    },
  ],
};
