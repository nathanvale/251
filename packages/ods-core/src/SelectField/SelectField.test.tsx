import React from "react";
import {
  fireEvent,
  queryByAttribute,
  render,
} from "@origin-digital/ods-testing-library";
import { SelectField } from "./SelectField";

test(`should render a SelectField`, () => {
  const { container } = render(
    <SelectField
      id="my-select-field"
      label="The title"
      helperText="This is the info"
      options={[
        { value: "syd", label: "Sydney" },
        { value: "mlb", label: "Melbourne" },
        { value: "prt", label: "Perth" },
      ]}
    />
  );

  expect(container).toMatchSnapshot();
});

test(`should render a SelectField with empty option`, () => {
  const { container } = render(
    <SelectField
      id="my-select-field"
      label="The title"
      helperText="This is the info"
      displayEmpty
      emptyAriaLabel="Empty"
      options={[
        { value: "syd", label: "Sydney" },
        { value: "mlb", label: "Melbourne" },
        { value: "prt", label: "Perth" },
      ]}
    />
  );

  expect(container).toMatchSnapshot();
});

test(`should render a SelectField with customised options`, () => {
  const { container } = render(
    <SelectField
      id="my-select-field"
      label="The title"
      helperText="This is the info"
      displayEmpty
      emptyAriaLabel="Empty"
    >
      <optgroup label="NSW">
        <option value="sydney">Sydney</option>
        <option value="newcastle">New Castle</option>
      </optgroup>
      <optgroup label="VIC">
        <option value="melbourne">Melbourne</option>
        <option value="bendigo">Bendigo</option>
      </optgroup>
    </SelectField>
  );

  expect(container).toMatchSnapshot();
});

test(`should render a SelectField options if both options and children props are provided`, () => {
  const { container } = render(
    <SelectField
      id="my-select-field"
      label="The title"
      helperText="This is the info"
      options={[
        { value: "cbr", label: "Canberra" },
        { value: "prt", label: "Perth" },
      ]}
    >
      <optgroup label="NSW">
        <option value="sydney">Sydney</option>
        <option value="newcastle">New Castle</option>
      </optgroup>
      <optgroup label="VIC">
        <option value="melbourne">Melbourne</option>
        <option value="bendigo">Bendigo</option>
      </optgroup>
    </SelectField>
  );

  const options = container.getElementsByTagName("option");
  expect(options).toHaveLength(2);
  expect(options.item(0)).toHaveTextContent("Canberra");
  expect(options.item(1)).toHaveTextContent("Perth");
  expect(container).toMatchSnapshot();
});

test(`should render a SelectField in error state`, () => {
  const { container } = render(
    <SelectField
      id="my-select-field"
      label="The title"
      helperText="This is the info"
      error
      options={[
        { value: "syd", label: "Sydney" },
        { value: "mlb", label: "Melbourne" },
        { value: "prt", label: "Perth" },
      ]}
    />
  );

  const label = queryByAttribute("data-id", container, "my-select-field-label");
  const helperText = queryByAttribute(
    "data-id",
    container,
    "my-select-field-helper-text"
  );

  expect(label).toHaveClass("Mui-error");
  expect(helperText).toHaveClass("Mui-error");
});

test("it passes down id and data-id", () => {
  const { getByTestId } = render(
    <SelectField
      data-id="select-field-base"
      id="my-select-field"
      options={[
        { value: "syd", label: "Sydney" },
        { value: "mlb", label: "Melbourne" },
        { value: "prt", label: "Perth" },
      ]}
    />
  );

  const SelectFieldRoot = getByTestId("select-field-base") as HTMLDivElement;
  const select = getByTestId("select-field-base-input") as HTMLSelectElement;

  expect(SelectFieldRoot).toBeDefined();
  expect(select?.tagName.toLowerCase()).toEqual("select");
});

test("when a select is entered onChanged is called with the value of input", () => {
  const onChange = jest.fn();
  const { getByTestId } = render(
    <SelectField
      id="my-select-field"
      onChange={onChange}
      displayEmpty
      emptyAriaLabel="Empty"
      options={[
        { value: "syd", label: "Sydney" },
        { value: "mlb", label: "Melbourne" },
        { value: "prt", label: "Perth" },
      ]}
    />
  );

  const select = getByTestId("my-select-field-input") as HTMLSelectElement;

  expect(select.value).toEqual("");

  fireEvent.change(select, { target: { value: "mlb" } });

  expect(onChange).toHaveBeenCalled();
  expect(select?.value).toEqual("mlb");
  expect(select?.selectedIndex).toEqual(2);
});
