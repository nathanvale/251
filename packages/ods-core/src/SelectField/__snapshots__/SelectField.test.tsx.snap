// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a SelectField 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-1 MuiFormControl-fullWidth"
    data-id="my-select-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
      data-id="my-select-field-label"
      data-shrink="true"
      for="my-select-field"
      id="my-select-field-label"
    >
      The title
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <select
        aria-describedby="my-select-field-helper-text"
        aria-invalid="false"
        autocomplete="false"
        class="MuiSelect-root MuiSelect-select MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
        data-id="my-select-field-input"
        id="my-select-field"
      >
        <option
          value="syd"
        >
          Sydney
        </option>
        <option
          value="mlb"
        >
          Melbourne
        </option>
        <option
          value="prt"
        >
          Perth
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained form-helper-text-grey400-colour-2 MuiFormHelperText-filled"
      data-id="my-select-field-helper-text"
      id="my-select-field-helper-text"
    >
      This is the info
    </p>
  </div>
</div>
`;

exports[`should render a SelectField options if both options and children props are provided 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-265 MuiFormControl-fullWidth"
    data-id="my-select-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
      data-id="my-select-field-label"
      data-shrink="true"
      for="my-select-field"
      id="my-select-field-label"
    >
      The title
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <select
        aria-describedby="my-select-field-helper-text"
        aria-invalid="false"
        autocomplete="false"
        class="MuiSelect-root MuiSelect-select MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
        data-id="my-select-field-input"
        id="my-select-field"
      >
        <option
          value="cbr"
        >
          Canberra
        </option>
        <option
          value="prt"
        >
          Perth
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained form-helper-text-grey400-colour-266 MuiFormHelperText-filled"
      data-id="my-select-field-helper-text"
      id="my-select-field-helper-text"
    >
      This is the info
    </p>
  </div>
</div>
`;

exports[`should render a SelectField with customised options 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-177 MuiFormControl-fullWidth"
    data-id="my-select-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-id="my-select-field-label"
      data-shrink="false"
      for="my-select-field"
      id="my-select-field-label"
    >
      The title
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <select
        aria-describedby="my-select-field-helper-text"
        aria-invalid="false"
        autocomplete="false"
        class="MuiSelect-root MuiSelect-select MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
        data-id="my-select-field-input"
        id="my-select-field"
      >
        <option
          aria-label="Empty"
          value=""
        />
        <optgroup
          label="NSW"
        >
          <option
            value="sydney"
          >
            Sydney
          </option>
          <option
            value="newcastle"
          >
            New Castle
          </option>
        </optgroup>
        <optgroup
          label="VIC"
        >
          <option
            value="melbourne"
          >
            Melbourne
          </option>
          <option
            value="bendigo"
          >
            Bendigo
          </option>
        </optgroup>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained form-helper-text-grey400-colour-178"
      data-id="my-select-field-helper-text"
      id="my-select-field-helper-text"
    >
      This is the info
    </p>
  </div>
</div>
`;

exports[`should render a SelectField with empty option 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-89 MuiFormControl-fullWidth"
    data-id="my-select-field"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-id="my-select-field-label"
      data-shrink="false"
      for="my-select-field"
      id="my-select-field-label"
    >
      The title
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <select
        aria-describedby="my-select-field-helper-text"
        aria-invalid="false"
        autocomplete="false"
        class="MuiSelect-root MuiSelect-select MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
        data-id="my-select-field-input"
        id="my-select-field"
      >
        <option
          aria-label="Empty"
          value=""
        />
        <option
          value="syd"
        >
          Sydney
        </option>
        <option
          value="mlb"
        >
          Melbourne
        </option>
        <option
          value="prt"
        >
          Perth
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained form-helper-text-grey400-colour-90"
      data-id="my-select-field-helper-text"
      id="my-select-field-helper-text"
    >
      This is the info
    </p>
  </div>
</div>
`;
